ObjectsAsAPI/ViewModels/OrderViewModel.cs

```diff
@@ -0,0 +1,54 @@
+﻿using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using ObjectsAsAPI.Models;
+using ObjectsAsAPI.Services;
+
+namespace ObjectsAsAPI.ViewModels;
+
+[QueryProperty("Order", nameof(Order))]
+public partial class OrderViewModel : BaseViewModel
+{
+    [ObservableProperty]
+    private Order _order = null!;
+
+    [RelayCommand]
+    public async Task CancelOrder()
+    {
+        var realm = RealmService.GetMainThreadRealm();
```

Comment:
In this method we first set the request to draft and only later we set it to pending, so that we can be sure that when the trigger is called (when `Status == Pending`), the payload has been uploaded already.
In a previous version we were creating the request directly after the dialog, with status set to pending. Sometimes this would cause issues on the backend, because the trigger couldn't find the payload yet.